main.go

```diff
@@ -86,6 +92,14 @@ func (gp Product) VerifyWithContext(ctx context.Context, key string) error {
 	}
 	defer resp.Body.Close()
 
+	// something on server side, should probably retry...
+	if resp.StatusCode >= 500 {
```

Comment:
The current code will retry the failed operation immediately up to 5 times. However, this may not be effective if there is a short-lived outage or issue with the external service. In such cases, immediately retrying multiple times in quick succession is unlikely to succeed.

A better approach would be to add a sleep between retry attempts, with an increasing backoff delay. This allows some time for the external service to potentially recover before the next retry attempt.

One option to implement this retry with backoff functionality is to use the backoff library, available at https://github.com/cenkalti/backoff.